### ISSUE TITLE
UI for logging drivers advanced mode

### ISSUE THREAD

**Issue by n313893254** [rancher/rancher#17229]
UI for https://github.com/rancher/rancher/issues/15558

**Comment by davidnuzik** [comment]
@aiwantaozi @cloudnautique Any chance either of you could point me to any resources regarding testing #15558 referenced in this issue? I am trying to validate the new "Edit as File" option in the UI currently and am having difficulty finding any docs around how to test with fluentd using this. I tried researching online and am having a hard time finding anything.
for example do not understand formatting and how to structure this
![image](https://user-images.githubusercontent.com/45179589/52077200-a86dfa00-254d-11e9-9be4-4b8b25516610.png)

**Comment by davidnuzik** [comment]
Version: Master 2.2 (2/21/19)
Confirmed issue. Note this was effectively tested in #17697 which has a comment describing the testing.

* Create custom logging target (edit as file)
* Input values for SSL Configuration section in the UI. I used fluentd for my custom target (@type forward)
* Inspect HTTP/API to ensure settings took (request payload and server resp and UI API).

Further testing involved here:
- Confirm we only allow types that we support in the UI.
- Cert paths listed (example below)
```
  client_key /fluentd/etc/config/ssl/cluster_c-hlwf7_client-key.pem
  client_cert /fluentd/etc/config/ssl/cluster_c-hlwf7_client-cert.pem
  tls_cert_path /fluentd/etc/config/ssl/cluster_c-hlwf7_ca.pem
```
- Certs are NOT validated when saving - Entered a separate issue for this here: #18310
- Validate HTTP to ensure looks correct.
